datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -233,7 +233,9 @@ macro_rules! compute_utf8_op_scalar {
             .as_any()
             .downcast_ref::<$DT>()
             .expect("compute_op failed to downcast left side array");
-        if let ScalarValue::Utf8(Some(string_value)) = $RIGHT {
+        if let ScalarValue::Utf8(Some(string_value))
+        | ScalarValue::LargeUtf8(Some(string_value)) = $RIGHT
+        {
```

Comment:
I'm not sure if we should add it.

Reply:
I definitely think we should add this -- it is required to evaluate the like expression with a scalar.